packages/amplify-util-uibuilder/src/utils/prePushHandler.ts

```diff
@@ -26,15 +23,23 @@ export const prePushHandler = async (context: $TSContext): Promise<void> => {
     ),
   ]);
 
-  if (!localSchema) {
+  if (
+    hasModels(localSchema)
+  ) {
+    printDetachedFormsWarning(formSchemas, localSchema);
+  } else {
     printer.debug('Local schema not found');
     return;
   }
-  printDetachedFormsWarning(formSchemas, localSchema);
 };
 
-const printDetachedFormsWarning = (formSchemas: { entities: AmplifyUIBuilder.Form[] }, localSchema: Schema): void => {
-  const modelNames = new Set(Object.keys(localSchema.models));
+const printDetachedFormsWarning = (formSchemas: { entities: AmplifyUIBuilder.Form[] }, localSchema: {models: unknown}): void => {
+  const models = localSchema.models;
+  if (!models || typeof models !== 'object') {
```

Comment:
nit: can't this use !hasModels helper?

Reply:
@lazpavel I talked with codegen and they will eventually be exporting types. So this is an intermediate update to remove the datastore import.